lib/bridge-client/index.js

```diff
@@ -73,6 +81,32 @@ BridgeClient.prototype._checkOptions = function(uri, options) {
   options.baseURI = options.baseURI || uri;
   options = merge(Object.create(BridgeClient.DEFAULTS), options);
   assert.ok(utils.validateLogger(options.logger), 'Invalid logger supplied');
+
+  return this._configureBlacklist(options);
+};
+
+/**
+ * Setup the store for our blacklist
+ * @private
+ */
+BridgeClient.prototype._configureBlacklist = function(options) {
+  // If we were not given a store, STORJ_TEMP or os.tmpdir. This is
+  // intentionally left out of BridgeClient.DEFAULTS since utils.tmpdir()
+  // accesses the filesystem, breaking storj-lib's compatibility with browsers
+  if(options.store === undefined) {
+    if(options.blacklistFolder === undefined) {
+      options.blacklistFolder = utils.tmpdir();
+    }
+    options.store = new FsBlobStore(options.blacklistFolder);
+  }
+
+  // Only test for the parts of abstract-blob-store that we need
+  assert.ok(typeof options.store.createWriteStream === 'function',
+    'Supplied store must implement abstract-blob-store');
+
+  assert.ok(typeof options.store.createReadStream === 'function',
+    'Supplied store must implement abstract-blob-store');
```

Comment:
We also need to check for `.exists()` which will be used in `UploadState.cleanup`